src/internal/document_api/mod.rs

```diff
@@ -435,53 +500,177 @@ pub fn encrypt_document<
     grant_to_author: bool,
     user_grants: &'a Vec<UserId>,
     group_grants: &'a Vec<GroupId>,
-    policy_grant: Option<&PolicyGrant>,
+    policy_grant: Option<&'a PolicyGrant>,
 ) -> impl Future<Item = DocumentEncryptResult, Error = IronOxideErr> + 'a {
     let (dek, doc_sym_key) = transform::generate_new_doc_key(recrypt);
     let doc_id = document_id.unwrap_or(DocumentId::goo_id(rng));
+    aes::encrypt_future(rng, &plaintext.to_vec(), *doc_sym_key.bytes())
+        .join(resolve_keys_for_grants(
+            auth,
+            user_grants,
+            group_grants,
+            policy_grant,
+            grant_to_author,
+        ))
+        .and_then(move |(encrypted_doc, (grants, key_errs))| {
+            encrypt_document_core(
+                auth,
+                recrypt,
+                dek,
+                encrypted_doc,
+                &doc_id,
+                grants(user_master_pub_key),
+            )
```

Comment:
This encrypts to the user master key even if you don't want to encrypt the document to the author?

Reply:
this was a partially applied function. I've encoded this in a more straightforward way now.